src/Library/Policy.hpp

```diff
@@ -20,19 +20,63 @@
 
 #include <Typedefs.hpp>
 #include <string>
+#include <vector>
+
+#include <RuleSet.hpp>
 
 namespace usbguard
 {
+  class Interface;
+
   class DLL_PUBLIC Policy
   {
     public:
       enum class EventType
       {
         Insert = 1,
         Update = 2,
-        Remove = 3,
+        Remove = 3
+      };
+
+      enum class SourceType
```

Comment:
Please remove this enum and make the Policy interface class take an instance of RuleSet instead. For example via a method names setRuleSet (in case there will be only one) or addRuleSet / appendRuleSet.

So that the usage is something like:


RuleSet ruleset;
ruleset.load(file);

Policy policy;
policy.setRuleSet(ruleset);
...
...

That way the Policy interface doesn't have to care about the source of the rule set.